internal/services/apicenter/api_center_environment_resource.go

```diff
@@ -0,0 +1,304 @@
+		},
+		"management_portal_uri": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPS,
+		},
+	}
+}
+
+func (r ApiCenterEnvironmentResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ApiCenterEnvironmentResource) ModelObject() interface{} {
+	return &ApiCenterEnvironmentResourceModel{}
+}
+
+func (r ApiCenterEnvironmentResource) ResourceType() string {
+	return "azurerm_api_center_environment"
+}
+
+func (r ApiCenterEnvironmentResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return environments.ValidateEnvironmentID
+}
+
+func (r ApiCenterEnvironmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			service, err := services.ParseServiceID(model.ServiceId)
+			if err != nil {
+				return err
+			}
+
+			// @favoretti: can't find any workspace creation buttons in the portal, assume everything defaults to "default" for now?
+			id := environments.NewEnvironmentID(subscriptionId, service.ResourceGroupName, service.ServiceName, "default", model.Identification)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing ApiCenter Environment %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			apiCenterEnvironmentProps := environments.EnvironmentProperties{
+				Kind:  environments.EnvironmentKind(model.Type),
+				Title: model.Name,
+			}
+
+			if model.Description != "" {
+				apiCenterEnvironmentProps.Description = pointer.To(model.Description)
+			}
+
+			apiCenterEnvironmentProps.Onboarding = &environments.Onboarding{}
+			if model.DevPortalUri != "" {
+				apiCenterEnvironmentProps.Onboarding.DeveloperPortalUri = pointer.To([]string{model.DevPortalUri})
+			}
+
+			if model.Instructions != "" {
+				apiCenterEnvironmentProps.Onboarding.Instructions = pointer.To(model.Instructions)
+			}
+
+			apiCenterEnvironmentProps.Server = &environments.EnvironmentServer{}
+
+			if model.ServerType != "" {
+				apiCenterEnvironmentProps.Server.Type = pointer.To(environments.EnvironmentServerType(model.ServerType))
+			}
+
+			if model.MgmtPortalUri != "" {
+				apiCenterEnvironmentProps.Server.ManagementPortalUri = pointer.To([]string{model.MgmtPortalUri})
+			}
+
+			apiCenterEnvironment := environments.Environment{
+				Name:       &model.Identification,
+				Properties: &apiCenterEnvironmentProps,
+				Type:       &model.Type,
```

Comment:
Please use the `pointer.To` function for these
```suggestion
				Name:       pointer.To(model.Identification),
				Properties: pointer.To(apiCenterEnvironmentProps),
				Type:       pointer.To(model.Type),
```